File: C++/프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <unordered_set>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> vec(n + 1, 1);
    
    for(int i = 2; i <= n; i++) {
        if(vec[i] == 1) {
            answer += 1;
            for(int j = 2; i * j <= n; j++) {
                vec[i * j] = 0;
            }
        }
    }
    
    return answer;
}

File: Python3/프로그래머스/1/12921. 소수 찾기/소수 찾기.py
import math


def solution(n):
    answer = [1] * (n + 1)
    answer[0], answer[1] = 0, 0
    
    for i in range(2, n + 1):
        if answer[i] == 1:
            j = 2
            while i * j <= n:
                answer[i * j] = 0
                j += 1
    
    return sum(answer)